webrtc.html

```diff
@@ -970,6 +970,83 @@
+        <var>p</var> are as follows:</p>
+        <ol>
+          <li>
+            <p>If the need for an identity assertion was identified when
+            <code>createOffer</code> was invoked, wait for <a href=
+            "#sec.identity-proxy-assertion-request">the identity assertion
+            request process</a> to complete.</p>
+          </li>
+          <li>
+            <p>If the identity provider was unable to produce an identity
+            assertion, reject <var>p</var> with a
+            <code>DOMException</code> object whose <code>name</code>
+            attribute has the value <code>NotReadableError</code>, and
+            abort these steps.</p>
+          </li>
+          <li>
+            <p>If <var>connection</var> was not constructed with a set of
+            certificates, and one has not yet been generated, wait for it
+            to be generated.</p>
+          </li>
+          <li>
+            <p>Inspect the system state to determine the currently
+            available resources as necessary for generating the offer, as
+            described in <span data-jsep="createoffer">[[!JSEP]]</span>.
+            </p>
+          </li>
+          <li>
+            <p>If this inspection failed for any reason, reject
+            <var>p</var> with a <code>DOMException</code> object whose
+            <code>name</code> attribute has the value
+            <code>OperationError</code>, and abort these steps.</p>
+          </li>
+          <li>
+            <p>Queue a task that runs the <a>final steps to create an
+            offer</a>, given <var>p</var>.</p>
+          </li>
+        </ol>
+        <p>The <dfn>final steps to create an offer</dfn> given a promise
+        <var>p</var> are as follows:</p>
+        <ol>
+          <li>
+            <p>If <var>connection</var>'s [[<a>isClosed</a>]] slot is
+            <code>true</code>, then abort these steps.</p>
+          </li>
+          <li>
+            <p>If <var>connection</var> was modified in such a way that
+            additional inspection of the system state is necessary, then
+            in parallel begin the <a>
+            steps to create an offer</a> again, given <var>p</var>, and
+            abort these steps.</p>
+
+            <div class="note">This may be necessary if, for example,
+            <code>createOffer</code> was called when only an audio
+            <code><a>RTCRtpTransceiver</a></code> was added to
+            <var>connection</var>, but while performing the <a>steps to
+            create an offer</a> in parallel, a video
+            <code><a>RTCRtpTransceiver</a></code> was added, requiring
+            additional inspection of video system resources.</div>
+          </li>
+          <li>
+            <p>Given the information that was obtained from previous
+            inspection, generate an SDP offer, <var>sdpString</var>, as
+            described in <span data-jsep="create-offer">[[!JSEP]]</span>.
+            </p>
```

Comment:
@taylor-b I see, and in the parallel steps it is less clear which state should be used. Arguably, this is an optimization since the loop-around catches any changes from initial set anyhow, but I see how it might be harder to write in the parallel steps in a way that's unambiguous. Provided JSEP lets us split things up this way, then what you have WFM.